frontend/src/components/TagFilter/index.tsx

```diff
@@ -0,0 +1,220 @@
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                selectMovie(!isMovieSelected);
+                            }} 
+                        />}
+                        label="Movies"
+                    />
+                    <FormControlLabel
+                        onClick={(event) => event.stopPropagation()}
+                        onFocus={(event) => event.stopPropagation()}
+                        control={
+                        <Checkbox
+                            name="book"
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                selectBook(!isBookSelected);
+                            }} 
+                        />}
+                        label="Books"
+                    />
+                </div>
+            </TypeSelector>
+            <div>
+                <Typography>
+                    Tags:
+                </Typography>
+                <div>
+                    {
+                        allTags.map(
+                            (tag) => (
+                                <FormControlLabel
+                                    onClick={(event) => event.stopPropagation()}
+                                    onFocus={(event) => event.stopPropagation()}
+                                    control={
+                                    <Checkbox 
+                                        name={tag}
+                                        onClick={(event) => {
+                                            event.stopPropagation();
+                                            if (allSelectedTags.includes(tag)) {
+                                                const index = allSelectedTags.indexOf(tag);
+                                                if (index > -1) {
+                                                    allSelectedTags.splice(index, 1)
+                                                }
+                                            } else {
+                                                selectTags([...allSelectedTags, tag]);
+                                            }
+                                        }} 
+                                    />}
+                                    label={tag.tagName}
+                                />
+                            )
+                        )
+                    }
+                </div>
+            </div>
+            <Button 
+                variant="contained"
+                onClick={()=>{
+                    setAllResults([]);
+                    getResultsByTag()
+                }}
+            >
+                Explore
+            </Button>
+        </div>
+    );
+    
+    return (
+        <div>
+            {
+                tabFilter
+            }                   
+            <SearchResultsInfiniteScroll
```

Comment:
maybe we can not render this when there are no results (e.g. allResults && allResults.length > 0 && <SearchResultsInfiniteScroll ...) so that we don't show the "Loading more results..." text when we aren't querying for anything, such as when we enter the page for the first time